### DISCUSSION TITLE
How to query for distinct attribute values?

### DISCUSSION THREAD

**Discussion by anandsubbu007** [appwrite/appwrite#4627] [Q&A]
after query some document in collection i only want unique values of one attribute.
like: unique brand name in product details with product in one category filter.

**Comment by TorstenDittmann** [comment]
Hey :)

Can you elaborate a little bit more and go into more detail on this issue?

**Comment by anandsubbu007** [comment]
after query some document in collection i only want unique values of one attribute.
like: unique brand name in product details with product in one category filter.

**Comment by christyjacob4** [comment]
@anandsubbu007 Thanks but it would help if you shared some examples of what you mean

**Comment by Zerdayne** [comment]
@christyjacob4 Hey, I'm not the creator of the issue but I can give an example how I need this feature.

I have a collection with some attributes. One of these attributes has the name "tags" and has the type string[].

What I need is the functionality to query this collection to get all distinct tags throughout the whole collection.

Some example documents (only the tags per document):
- weapon, sword, short
- weapon, sword, long
- armor, iron

Expected result:
- weapon
- sword
- short
- long
- armor
- iron

Is that example helping you?

**Comment by stnguyen90** [comment]
The recommended approach for doing something like this is to maintain a metadata collection and update the metadata with every document create/update/delete event. Then, you can fetch that metadata collection/document to get the unique values.
